src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -28,41 +38,357 @@ const ActorInit Bg_Ingate_InitVars = {
+                } else {
+                    gSaveContext.eventInf[4] |= 1;
+                }
+                this->actionFunc = func_809542A0;
+            }
+        } else if ((ActorCutscene_GetCurrentIndex() == -1) && (this->unk164 != 0)) {
+            // This sound macro doesn't exist??
+            Actor_PlaySfxAtPos(&this->dyna.actor, 0x211E);
+            func_80953BEC(this);
+        }
+    }
+    if (ActorCutscene_GetCurrentIndex() != this->unk16E) {
+        if (ActorCutscene_GetCurrentIndex() != -1) {
+            func_800DFAC8(camera, 1);
+            player->stateFlags1 |= ACTOR_FLAG_20;
+            globalCtx->actorCtx.unk5 &= ~0x4;
+        } else {
+            func_800DFAC8(camera, 0x2F);
+            player->stateFlags1 &= ~ACTOR_FLAG_20;
+        }
+    }
+    this->unk16E = ActorCutscene_GetCurrentIndex();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953DA8.s")
+void func_809541B8(BgIngate* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953E38.s")
+    if (this->unk160 & 0x4) {
+        if ((player->transformation == PLAYER_FORM_HUMAN) && (player->actor.bgCheckFlags & 1) &&
+            (this->dyna.actor.xzDistToPlayer < 40.0f)) {
+            if (this->dyna.actor.playerHeightRel > 15.0f) {
+                func_800B7298(globalCtx, &this->dyna.actor, 7);
+                this->dyna.actor.textId = 0x9E6;
+                func_801518B0(globalCtx, this->dyna.actor.textId, NULL);
+                this->actionFunc = func_809543D4;
+            }
+        }
+    } else if (DynaPolyActor_IsInRidingMovingState(&this->dyna) == 0) {
+        this->unk160 |= 4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953EA4.s")
+void func_809542A0(BgIngate* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.eventInf[5] & 1) != 0) {
+        globalCtx->nextEntranceIndex = 0xA820;
+        gSaveContext.eventInf[5] &= 0xFE;
+    } else {
+        globalCtx->nextEntranceIndex = 0xA810;
+    }
+    gSaveContext.nextCutsceneIndex = 0;
+    globalCtx->sceneLoadFlag = 0x14;
+    globalCtx->unk_1887F = 3;
+    gSaveContext.nextTransition = 3;
+    this->actionFunc = func_80953F8C;
+    gSaveContext.weekEventReg[90] &= 0xBF;
+    func_800FE498();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F14.s")
+void func_80954340(BgIngate* this, GlobalContext* globalCtx) {
+    if (!DECR(this->unk16A)) {
+        if (this->unk164 != 0) {
+            func_800B7298(globalCtx, &this->dyna.actor, 6U);
+            this->unk164 = &globalCtx->setupPathList[this->unk164->unk1];
+            func_80953F14(this, globalCtx);
+            func_800FE484();
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ingate/func_80953F8C.s")
+void func_809543D4(BgIngate* this, GlobalContext* globalCtx) {
+    u8 talkState;
```

Comment:
```suggestion
    u8 talkState = Message_GetState(&globalCtx->msgCtx);
```
Would also delete the `talkState = Message_GetState(&globalCtx->msgCtx);` below.